src/nifake/tests/test_session.py

```diff
@@ -328,3 +328,32 @@ def test_multiple_return_params(self):
             assert self.patched_errors.handle_error.call_count == 2
             self.patched_errors.handle_error.assert_called_with(session, self.patched_library.niFake_GetCalDateAndTime.return_value)
     '''
+
+    def test_get_vi_int32_attribute(self):
+        self.patched_library.niFake_GetAttributeViInt32.side_effect = self.side_effects_helper.niFake_GetAttributeViInt32
+        test_number = 3
+        self.side_effects_helper['GetAttributeViInt32']['attributeValue'] = test_number
+        with nifake.Session('dev1') as session:
+            attr_int = session.read_write_integer
+            assert(attr_int == test_number)
+            self.patched_library.niFake_GetAttributeViInt32.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', 1000004, ANY)
+
+    def test_set_vi_int32_attribute(self):
+        self.patched_library.niFake_SetAttributeViInt32.side_effect = self.side_effects_helper.niFake_SetAttributeViInt32
+        attribute_id = 1000004
+        test_number = 1
+        with nifake.Session('dev1') as session:
+            session.read_write_integer = test_number
+            self.patched_library.niFake_SetAttributeViInt32.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', attribute_id, 1)
+
+    def test_error_on_close(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_library.niFake_close.side_effect = self.side_effects_helper.niFake_close
+        self.side_effects_helper['close']['return'] = test_error_code
+        self.side_effects_helper['close']['vi'] = SESSION_NUM_FOR_TEST
```

Comment:
What is this supposed to accomplish? niFake_Close() takes the ViSession as an input. You should be checking that with assert_called_once_with...